### PR TITLE
Fwmt 2546 accommodate hard refusal

### PR DESCRIPTION
**Pull request by SteveScorfield**

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/HardRefusalReceivedProcessor.java [hunk 1]
```diff
@@ -33,15 +40,46 @@
...
+
     UUID caseId = (caseIdHolder != null) ? caseIdHolder : outcome.getCaseId();
 
     gatewayEventManager.triggerEvent(String.valueOf(caseId), PROCESSING_OUTCOME,
     "survey type", type,
     "processor", "HARD_REFUSAL_RECEIVED",
     "original caseId", String.valueOf(outcome.getCaseId()),
     "Site Case id", (outcome.getSiteCaseId() != null ? String.valueOf(outcome.getSiteCaseId()) : "N/A"));
+    if (refusalCodes != null) {
+      if (outcome.getRefusal() != null && (type.equals("HH") || type.equals("NC"))) {
```
[the remark this answers is not in the document]
Comment (ocallaghanit) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/HardRefusalReceivedProcessor.java:69]: Might ge good to try and simplify / extract this entire block of code into a new method

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/NewUnitAddressLinkedProcessor.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public UUID process(OutcomeSuperSetDto outcome, UUID caseIdHolder, String type)
     root.put("sourceCase", "NEW_UNIT");
     root.put("outcome", outcome);
     root.put("newCaseId", caseId);
-    root.put("sourceCaseId", outcome.getSiteCaseId());
+    root.put("sourceCaseId", "9170085a-1367-4270-8674-acc6d72671e1");
```
Comment (ocallaghanit) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/NewUnitAddressLinkedProcessor.java:61]: Hard coded case id?

**Review by ocallaghanit — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/NewUnitAddressLinkedProcessor.java (hunk 2 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/impl/NewUnitAddressLinkedProcessor.java:61]: This has been updated. Not sure how this even got there!

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/RefusalEncryptionLookup.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.ons.census.fwmt.outcomeservice.converter;
+
+import java.util.ArrayList;
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/RefusalEncryptionLookup.java:3]: Been removed.

**Review by ocallaghanit — APPROVED** [review]

**Pull request merged**